compiler/rustc_query_impl/src/plumbing.rs

```diff
@@ -788,3 +788,64 @@ macro_rules! define_queries {
         }
     }
 }
+
+pub fn print_query_stack<'tcx>(
+    qcx: QueryCtxt<'tcx>,
+    mut current_query: Option<QueryJobId>,
+    handler: &Handler,
+    num_frames: Option<usize>,
+) -> usize {
+    // Be careful relying on global state here: this code is called from
+    // a panic hook, which means that the global `Handler` may be in a weird
+    // state if it was responsible for triggering the panic.
+    let mut i = 0;
+
+    #[cfg(not(parallel_compiler))]
+    let query_map = qcx.try_collect_active_jobs();
+
+    #[cfg(parallel_compiler)]
+    let query_map = rustc_middle::ty::tls::with_context(|context| {
+        use std::sync::mpsc::channel;
+        use std::thread;
+        use std::time::Duration;
+
+        let timeout = Duration::from_secs(5); // Set the timeout to 5 seconds
+
+        thread::scope(|s| {
+            let (tx, rx) = channel();
+
+            s.spawn(move || {
+                rustc_middle::ty::tls::enter_context(context, || {
+                    let query_map = qcx.try_collect_active_jobs();
```

Comment:
Calling `try_collect_active_jobs` on an outside thread can run into issues with `WorkerLocal`.

Reply:
Reporting ice is an action outside the thread pool. This is not a regression. `try_collect_active_jobs` should not access `WorkerLocal`. Even if it is accessed due to future modifications, it will panic, at least not into a deadlock.

Reply:
I used rayon_core::join instead of spawn so that there is no problem with WorkerLocal.

But I still have a bit doubts—— if reporting ice is running in rayon's thread pool, why the deadlock handler is not performed